### PR TITLE
added infolabels processing

### PR DESCRIPTION
**Pull request by elgatito**
Moved cache setter isndie of a getter function.

Added resolvers for /show  and /movie.

### CODE REVIEW

**Comment by elgatito** [comment]
Rest of functionality to find information in Kodi database will be written in addon's python part. 
Should be setted on the first call to navigation.py, when setted, all further calls from navigation and the daemon will successfully fetch infolabels from Kodi.

**Comment by elgatito** [comment]
What we have:
- If currently playing item can be identified - get it, make ListItem, save to Window.Properties and result to navigation.py
- If playing item launched from search or whatever (no /movie/*tmdb*  or /show/*tmdb available) we try to get information from ListItem
- navigation.py is launched twice, first time with a url saved in strm file, second with "/play" url. First time we can resolve item, so we do that and save in properties, second request reads properties.

File: xbmc/types.go [hunk 1]
```diff
@@ -88,13 +89,14 @@ type ListItemInfo struct {
...
-	Thumbnail string `json:"thumb,omitempty"`
-	Poster    string `json:"poster,omitempty"`
-	Banner    string `json:"banner,omitempty"`
-	FanArt    string `json:"fanart,omitempty"`
-	ClearArt  string `json:"clearart,omitempty"`
-	ClearLogo string `json:"clearlogo,omitempty"`
-	Landscape string `json:"landscape,omitempty"`
+	Thumbnail 	 string `json:"thumb,omitempty"`
+	Poster    	 string `json:"poster,omitempty"`
+	TvshowPoster string `json:"tvshowposter,omitempty"`
```
Comment (scakemyer) [inline xbmc/types.go]: `TvShowPoster` and indentation

File: api/infolabels.go [hunk 2]
```diff
@@ -0,0 +1,190 @@
+package api
+
+import (
+    "strconv"
+    "errors"
+    "encoding/json"
+    "strings"
```
Comment (scakemyer) [inline api/infolabels.go]: Order should be: errors, strings, strconv then encoding/json

File: api/infolabels.go [hunk 3]
```diff
@@ -0,0 +1,190 @@
...
+    return string(b)
+}
+
+func convertItemToMap(item *xbmc.ListItem) map[string]string {
+    b, _ := json.Marshal(item)
+    m := make(map[string]string)
+    json.Unmarshal(b, &m)
+
+    return m
+}
```
Comment (scakemyer) [inline api/infolabels.go]: Just use `json.Marshal`, `string()` and `make()` directly, no need for those wrapper methods. `convertItemToMap` isn't being used either.

File: api/infolabels.go [hunk 4]
```diff
@@ -0,0 +1,190 @@
...
+        ctx.String(200, labels)
+    }
+}
+
+func InfoLabelsShow(btService *bittorrent.BTService) gin.HandlerFunc {
+    return func(ctx *gin.Context) {
+        tmdbId := ctx.Params.ByName("showId")
+
+        show := tmdb.GetShowById(tmdbId, "")
+        labels := GetShowInfoLabels(show)
```
Comment (scakemyer) [inline api/infolabels.go]: `GetShowInfoLabels` is only used here, no need for it to be exported, but might as well just put its code here instead and get rid of it. Same for the other ones.

File: api/infolabels.go [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+                key := strings.Replace(k, "ListItem.", "", 1)
+                listItemLabels[key] = v
+            }
+
+            b, _ := json.Marshal(listItemLabels)
+            labelsString = string(b)
+            saveEncoded(labelsString)
+        } else if encoded := xbmc.GetWindowProperty("ListItem.Encoded"); len(encoded) > 0 {
+            labelsString = encoded
+        }
```
Comment (scakemyer) [inline api/infolabels.go]: extraneous whitespace

File: api/infolabels.go [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+func InfoLabelsShow(btService *bittorrent.BTService) gin.HandlerFunc {
+    return func(ctx *gin.Context) {
+        tmdbId := ctx.Params.ByName("showId")
+
+        show := tmdb.GetShowById(tmdbId, "")
+        item := show.ToListItem()
+
+        saveEncoded(encodeItem(item))
+
+        ctx.Writer.Header().Set("Access-Control-Allow-Origin", "*")
```
Comment (scakemyer) [inline api/infolabels.go]: You can remove those, we're not calling this from the WebUI.

**Pull request merged**